### PR TITLE
Add shopify aa1 to master

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no Express Node.js application.

**What is the new behavior (if this is a feature change)?**
- The connector Express Node.js application has been created. It is already able to deliver data to our frontend in this state. @rhufsky can be proud now.
- The application now uses username / password authentication.
- The application is now able to fetch data from Shopify and deliver it through a Promise to our React Frontend.

**Other information**:
- The version has been set to `v1.0.0` and marks the first release of our application.
- `http://localhost:3000` is still being whitelisted for CORS right now. This will change, as soon as we deploy our application to improve security.

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    res.status(401).send(err);
+  } else if (err.message == "Not allowed by CORS") {
+    res.status(401).send("Not allowed from this origin.");
+  } else {
+    next(err);
+  }
+}
+
+// Whitelisted domains
+let whitelist = ["https://kaffeerudel.at", "http://localhost:3000"];
```
Comment (Theralun) [inline config/index.js:17]: constable goodie, what in the name of the queen are you doing?

File: services/authentication.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  } catch (err) {
+    return null;
+  }
+}
+
+router.post("/", (req, res) => {
+  res.setHeader("Access-Control-Allow-Methods", "POST");
+  if (req.body.hasOwnProperty("refreshToken")) {
+    // check if user uses refresh token
+    let token = checkRefreshToken(req.body.refreshToken);
```
Comment (Theralun) [inline services/authentication.js:32]: Const?

File: services/authentication.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if (token != null) {
+      let newToken = jwt.sign(
```
Comment (Theralun) [inline services/authentication.js:34]: Const?

File: services/authentication.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      });
+    }
+  } else {
+    // if not refresh token use username and password
+    const { username, password } = req.body;
+    if (
+      username == process.env.AUTH_USERNAME &&
+      password == process.env.AUTH_PASSWORD
+    ) {
+      let token = jwt.sign({ id: user.id, username: user.username }, sec1, {
```
Comment (Theralun) [inline services/authentication.js:59]: Const?

File: services/authentication.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        expiresIn: config.durationToken,
+      });
+      let refreshToken = jwt.sign(
```
Comment (Theralun) [inline services/authentication.js:62]: Const?

File: services/shopify.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+//#region > Functions
+// Get all orders
+function getOrders() {
+  return new Promise((resolve, reject) => {
+    let apiKey = process.env.SHOPIFY_API_KEY;
+    let apiPass = process.env.SHOPIFY_API_PASS;
+    let shopname = "bluelupi";
+    let apiVersion = "2020-07";
+    let resource = "orders";
```
Comment (Theralun) [inline services/shopify.js:14]: Whole lotta Const

File: services/shopify.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+// Get details by order id
+function getOrderDetails(orderId) {
+  return new Promise((resolve, reject) => {
+    let apiKey = process.env.SHOPIFY_API_KEY;
+    let apiPass = process.env.SHOPIFY_API_PASS;
+    let shopname = "bluelupi";
+    let apiVersion = "2020-07";
+    let resource = "orders";
```
Comment (Theralun) [inline services/shopify.js:36]: Whole lotta const?

File: config/index.js (hunk 1 above)
Comment (Aichnerc) [reply config/index.js:17]: Oh thanks for commenting on this part of the code.

I did actually copy this code from another repository and didn't realise it is offensive.
The new official term for `whitelist` ist `allowlist`.

```suggestion
const allowlist = ["https://kaffeerudel.at", "http://localhost:3000"];
```

**Pull request merged**